File: swiftpm/Sources/NumbersLib/NumbersLib.swift
import Foundation
import BigInt

public func fibonacci(n: Int) -> BigInt {
  var firstTerm = BigInt(0)
  var secondTerm = BigInt(1)

  for _ in 0..<n {
    let temp = firstTerm
    firstTerm = secondTerm
    secondTerm += temp
  }

  return firstTerm
}

public func fibonacciRandom() -> String {
  let randomNum = Int.random(in: 1...50)
  let fib = fibonacci(n: randomNum)

  return "Fib[\(randomNum)] = \(fib)"
}

public func factorial(n: Int) -> BigInt {
  var fact = BigInt(1)

  for i in 1...n {
    fact *= BigInt(i)
  }

  return fact
}

public func factorialRandom() -> String {
  let randomNum = Int.random(in: 1...25)
  let fact = factorial(n: randomNum)

  return "Fact[\(randomNum)] = \(fact)"
}

File: dotnet/modules/Numbers.Utils/Numbers.Utils.Lib/NumbersLib.cs
using System.Numerics;

namespace Numbers.Lib;

public class NumbersLib
{
    public static BigInteger Fibonacci(int n)
    {
        BigInteger firstTerm = 0;
        BigInteger secondTerm = 1;

        for (int i = 0; i < n; i++)
        {
            BigInteger temp = firstTerm;
            firstTerm = secondTerm;
            secondTerm += temp;
        }

        return firstTerm;
    }

    public static string FibonacciRandom()
    {
        Random rng = new Random();
        int randomNum = rng.Next(1, 51);

        BigInteger fib = Fibonacci(randomNum);

        return $"Fib[{randomNum}] = {fib}";
    }

    public static BigInteger Factorial(int n)
    {
        BigInteger fact = 1;

        for (int i = 1; i <= n; i++)
        {
            fact *= i;
        }

        return fact;
    }

    public static string FactorialRandom()
    {
        Random rng = new Random();
        int randomNum = rng.Next(1, 26);

        BigInteger fact = Factorial(randomNum);

        return $"Fact[{randomNum}] = {fact}";
    }
}
